democracy/tests/test_comment.py

```diff
@@ -685,3 +704,83 @@ def test_only_unauthenticated_can_set_author_name(client, can_set_author_name, r
     else:
         assert response.status_code == 403
         assert 'Authenticated users cannot set author name.' in response.data['status']
+
+
+@pytest.mark.django_db
+def test_comment_put(john_doe_api_client, default_hearing, get_detail_url):
+    section = default_hearing.get_main_section()
+    comment = section.comments.all()[0]
+    comment_data = get_comment_data(content='updated content', section=section.id)
+    url = get_detail_url(comment)
+
+    response = john_doe_api_client.put(url, data=comment_data)
+    assert response.status_code == 200
+
+    comment.refresh_from_db()
+    assert comment.content == 'updated content'
+
+
+@pytest.mark.django_db
+def test_comment_patch(john_doe_api_client, default_hearing, get_detail_url):
+    section = default_hearing.get_main_section()
+    comment = section.comments.all()[0]
+    url = get_detail_url(comment)
+
+    response = john_doe_api_client.patch(url, data={'content': 'updated content'})
+    assert response.status_code == 200
+
+    comment.refresh_from_db()
+    assert comment.content == 'updated content'
+
+
+@pytest.mark.django_db
+def test_cannot_modify_others_comments(jane_doe_api_client, default_hearing, get_detail_url):
+    section = default_hearing.get_main_section()
+    comment = section.comments.all()[0]
+    old_content = comment.content
+    comment_data = get_comment_data(content='updated content', section=section.id)
+    url = get_detail_url(comment)
+
+    response = jane_doe_api_client.put(url, data=comment_data)
+    assert response.status_code == 403
+
+    comment.refresh_from_db()
+    assert comment.content == old_content
+
+
+@pytest.mark.django_db
+def test_comment_delete(john_doe_api_client, default_hearing, get_detail_url):
+    comment = default_hearing.get_main_section().comments.all()[0]
+    url = get_detail_url(comment)
+
+    response = john_doe_api_client.delete(url)
+    assert response.status_code == 204
+
+    comment = SectionComment.objects.everything().get(id=comment.id)
+    assert comment.deleted is True
+
+
```

Comment:
Here, we should add a test that anonymous comments cannot be deleted by any user, and any comments cannot be deleted by an anonymous user.